### PR TITLE
Anko - Handle logic in request invite screen

### PR DESCRIPTION
**Pull request by at-ngoctran**
#### Related issues (Required)
- Task #156 

#### Description (Required)
- [x] Handle logic in show request invite

### CODE REVIEW

File: app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt [hunk 1]
```diff
@@ -64,5 +64,7 @@ class GroupRepository : GroupDataSource {
 
     override fun acceptInvite(userId: String, invite: Invite): Single<Boolean> = remoteDataSource.acceptInvite(userId, invite)
 
-    override fun getUserInfo(userId: String): Single<User> = remoteDataSource.getUserInfo(userId)
+    override fun getUserInfo(groupId: String): Observable<User> {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/project-way/blob/486a3918b16c1430859211b051956239bdb84f6a/app/src/main/java/vn/asiantech/way/data/source/GroupRepository.kt#L67"></span></td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt [hunk 2]
```diff
@@ -6,23 +6,148 @@ import android.view.View
...
+    }
+
+    /**
+     * On click button Cancel of list request
+     */
+    internal fun eventOnButtonCancelClick(userId: String) {
+        handleRemoveRequestInGroup(userId)
+    }
+
+    /**
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/project-way/blob/486a3918b16c1430859211b051956239bdb84f6a/app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt#L87"></span>Comments for private functions should be avoided. Prefer giving the function an expressive name. Split it up in smaller, self-explaining functions if necessary.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt [hunk 3]
```diff
@@ -6,23 +6,148 @@ import android.view.View
...
+     * On handle  after get request list of user completed
+     */
+    private fun handleGetRequestsOfUserSuccess(user: User) {
+        requestsUser.clear()
+        requestsUser.add(user)
+        adapter.notifyItemInserted(requestsUser.size - 1)
+    }
+
+    /**
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/project-way/blob/486a3918b16c1430859211b051956239bdb84f6a/app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt#L96"></span>Comments for private functions should be avoided. Prefer giving the function an expressive name. Split it up in smaller, self-explaining functions if necessary.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt [hunk 4]
```diff
@@ -6,23 +6,148 @@ import android.view.View
...
+     * On handle  after get request list of user failed
+     */
+    private fun handleGetRequestsOfUserFailed(error: Throwable) {
+        toast(error.message.toString())
+    }
+
+    /**
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/project-way/blob/486a3918b16c1430859211b051956239bdb84f6a/app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt#L103"></span>Comments for private functions should be avoided. Prefer giving the function an expressive name. Split it up in smaller, self-explaining functions if necessary.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt [hunk 5]
```diff
@@ -6,23 +6,148 @@ import android.view.View
...
+     * On handle  after add user to group success
+     */
+    private fun handleAddUserToGroupSuccess(user: User) {
+        handleRemoveRequestInGroup(user.id)
+    }
+
+    /**
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/project-way/blob/486a3918b16c1430859211b051956239bdb84f6a/app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt#L110"></span>Comments for private functions should be avoided. Prefer giving the function an expressive name. Split it up in smaller, self-explaining functions if necessary.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt [hunk 6]
```diff
@@ -6,23 +6,148 @@ import android.view.View
...
+     * On handle  after add user to group failed
+     */
+    private fun handleAddUserToGroupFailed(error: Throwable) {
+        toast(error.message.toString())
+    }
+
+    /**
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/project-way/blob/486a3918b16c1430859211b051956239bdb84f6a/app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt#L117"></span>Comments for private functions should be avoided. Prefer giving the function an expressive name. Split it up in smaller, self-explaining functions if necessary.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt [hunk 7]
```diff
@@ -6,23 +6,148 @@ import android.view.View
...
+    private fun handleRemoveRequestInGroup(userId: String) {
+        addDisposables(
+                viewModel
+                        .removeRequestInGroup(userId, groupId)
+                        .observeOnUiThread()
+                        .subscribe(this::handleRemoveRequestInGroupSuccess, this::handleRemoveRequestInGroupFailed)
+        )
+    }
+
+    /**
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/project-way/blob/486a3918b16c1430859211b051956239bdb84f6a/app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt#L129"></span>Comments for private functions should be avoided. Prefer giving the function an expressive name. Split it up in smaller, self-explaining functions if necessary.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt [hunk 8]
```diff
@@ -6,23 +6,148 @@ import android.view.View
...
+     * On handle after emove request in group success
+     */
+    private fun handleRemoveRequestInGroupSuccess(isSuccess: Boolean) {
+        toast(R.string.success)
+    }
+
+    /**
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/project-way/blob/486a3918b16c1430859211b051956239bdb84f6a/app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt#L136"></span>Comments for private functions should be avoided. Prefer giving the function an expressive name. Split it up in smaller, self-explaining functions if necessary.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt [hunk 9]
```diff
@@ -6,23 +6,148 @@ import android.view.View
...
+     * On handle after emove request in group success
+     */
+    private fun handleRemoveRequestInGroupFailed(error: Throwable) {
+        toast(error.message.toString())
+    }
+
+    /**
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/project-way/blob/486a3918b16c1430859211b051956239bdb84f6a/app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt#L143"></span>Comments for private functions should be avoided. Prefer giving the function an expressive name. Split it up in smaller, self-explaining functions if necessary.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestViewModel.kt [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+/**
+ * Request View Model
+ * @author NgocTTN
+ */
+class ShowRequestViewModel(private val wayRepository: WayRepository, private val groupRepository: GroupRepository) {
+    internal var progressDialogObservable: BehaviorSubject<Boolean> = BehaviorSubject.create()
+
+    constructor() : this(WayRepository(), GroupRepository())
+
+    internal fun getRequestsOfUser(groupId: String): Observable<User> {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestViewModel.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/project-way/blob/486a3918b16c1430859211b051956239bdb84f6a/app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestViewModel.kt#L20"></span></td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestViewModel.kt [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+        return groupRepository
+                .getUserInfo(groupId)
+    }
+
+    internal fun addUserToGroup(groupId: String, userId: String): Observable<User> {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestViewModel.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/project-way/blob/486a3918b16c1430859211b051956239bdb84f6a/app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestViewModel.kt#L25"></span></td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestViewModel.kt [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+        return wayRepository.addUserToGroup(userId, BodyAddUserToGroup(groupId))
+                .doOnSubscribe { progressDialogObservable.onNext(true) }
+                .doFinally { progressDialogObservable.onNext(false) }
+    }
+
+    internal fun removeRequestInGroup(groupId: String, userId: String): Single<Boolean> {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestViewModel.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/project-way/blob/486a3918b16c1430859211b051956239bdb84f6a/app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestViewModel.kt#L31"></span></td>
    </tr>
  </tbody>
</table>

**Comment by at-dev-support-01** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please rebase to get rid of the merge commits in this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt (hunk 2 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/project-way/blob/4463240b07d49e77c97e49f7a651b24f03b8f3b0/app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt#L87"></span>Comments for private functions should be avoided. Prefer giving the function an expressive name. Split it up in smaller, self-explaining functions if necessary.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt [hunk 13]
```diff
@@ -6,23 +6,148 @@ import android.view.View
...
+     * On handle after get request list of user completed
+     */
+    private fun handleGetRequestsOfUserSuccess(user: User) {
+        requestsUser.clear()
+        requestsUser.add(user)
+        adapter.notifyItemInserted(requestsUser.size - 1)
+    }
+
+    /**
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/project-way/blob/4463240b07d49e77c97e49f7a651b24f03b8f3b0/app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt#L96"></span>Comments for private functions should be avoided. Prefer giving the function an expressive name. Split it up in smaller, self-explaining functions if necessary.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt [hunk 14]
```diff
@@ -6,23 +6,148 @@ import android.view.View
...
+     * On handle after get request list of user failed
+     */
+    private fun handleGetRequestsOfUserFailed(error: Throwable) {
+        toast(error.message.toString())
+    }
+
+    /**
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/project-way/blob/4463240b07d49e77c97e49f7a651b24f03b8f3b0/app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt#L103"></span>Comments for private functions should be avoided. Prefer giving the function an expressive name. Split it up in smaller, self-explaining functions if necessary.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt [hunk 15]
```diff
@@ -6,23 +6,148 @@ import android.view.View
...
+     * On handle after add user to group success
+     */
+    private fun handleAddUserToGroupSuccess(user: User) {
+        handleRemoveRequestInGroup(user.id)
+    }
+
+    /**
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/project-way/blob/4463240b07d49e77c97e49f7a651b24f03b8f3b0/app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt#L110"></span>Comments for private functions should be avoided. Prefer giving the function an expressive name. Split it up in smaller, self-explaining functions if necessary.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt [hunk 16]
```diff
@@ -6,23 +6,148 @@ import android.view.View
...
+     * On handle after add user to group failed
+     */
+    private fun handleAddUserToGroupFailed(error: Throwable) {
+        toast(error.message.toString())
+    }
+
+    /**
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/project-way/blob/4463240b07d49e77c97e49f7a651b24f03b8f3b0/app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt#L117"></span>Comments for private functions should be avoided. Prefer giving the function an expressive name. Split it up in smaller, self-explaining functions if necessary.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt [hunk 17]
```diff
@@ -6,23 +6,148 @@ import android.view.View
...
+    private fun handleRemoveRequestInGroup(userId: String) {
+        addDisposables(
+                viewModel
+                        .removeRequestInGroup(userId, groupId)
+                        .observeOnUiThread()
+                        .subscribe(this::handleRemoveRequestInGroupSuccess, this::handleRemoveRequestInGroupFailed)
+        )
+    }
+
+    /**
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/project-way/blob/4463240b07d49e77c97e49f7a651b24f03b8f3b0/app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt#L129"></span>Comments for private functions should be avoided. Prefer giving the function an expressive name. Split it up in smaller, self-explaining functions if necessary.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt [hunk 18]
```diff
@@ -6,23 +6,148 @@ import android.view.View
...
+     * On handle after emove request in group success
+     */
+    private fun handleRemoveRequestInGroupSuccess(isSuccess: Boolean) {
+        toast(R.string.success)
+    }
+
+    /**
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/project-way/blob/4463240b07d49e77c97e49f7a651b24f03b8f3b0/app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt#L136"></span>Comments for private functions should be avoided. Prefer giving the function an expressive name. Split it up in smaller, self-explaining functions if necessary.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt [hunk 19]
```diff
@@ -6,23 +6,148 @@ import android.view.View
...
+     * On handle after emove request in group success
+     */
+    private fun handleRemoveRequestInGroupFailed(error: Throwable) {
+        toast(error.message.toString())
+    }
+
+    /**
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/project-way/blob/4463240b07d49e77c97e49f7a651b24f03b8f3b0/app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt#L143"></span>Comments for private functions should be avoided. Prefer giving the function an expressive name. Split it up in smaller, self-explaining functions if necessary.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/vn/asiantech/way/data/source/remote/GroupRemoteDataSource.kt [hunk 20]
```diff
@@ -284,7 +284,27 @@ class GroupRemoteDataSource : GroupDataSource {
         return result
     }
 
-    override fun getUserInfo(userId: String): Single<User> = HypertrackApi.instance.getUserInfo(userId)
+    override fun getUserInfo(groupId: String): Observable<User> {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/data/source/remote/GroupRemoteDataSource.kt]: Please fix this warning

File: app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt [hunk 21]
```diff
@@ -6,23 +6,124 @@ import android.view.View
...
+            return instance
+        }
+    }
+
     private lateinit var ui: ShowRequestFragmentUI
-    private var users: MutableList<User> = ArrayList()
+    private lateinit var adapter: RequestAdapter
+    private lateinit var viewModel: ShowRequestViewModel
+    private lateinit var groupId: String
+    private var requestsUser = mutableListOf<User>()
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/group/request/ShowRequestFragment.kt]: `requestsUser` change to `users`

**Comment by at-vinhhuynh** [comment]
@at-tienhoang 
I finish review this PR, please check it again.
thank you!

**Pull request merged**